### PR TITLE
Fix #368

### LINKED ISSUE #368: `A4-7-1`:  `/=` is incorrectly reported
### Affected rules
 - `A4-7-1`

### Description

`/=` is incorrectly reported as causing data loss regardless of the arguments involved.

### Example

```cpp
i /= 2;
```

### PR DESCRIPTION
**Pull request by rvermeulen**
## Description

Fix FP reported in #368.

## Change request type

 - [ ] Release or process automation (GitHub workflows, internal scripts)
 - [ ] Internal documentation
 - [ ] External documentation
 - [x] Query files (`.ql`, `.qll`, `.qls` or unit tests)
 - [ ] External scripts (analysis report or other code shipped as part of a release)

## Rules with added or modified queries

 - [ ] No rules added
 - [ ] Queries have been added for the following rules:
     - _rule number here_
 - [x] Queries have been modified for the following rules:
     - A4-7-1

## Release change checklist

A change note ([development_handbook.md#change-notes](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#change-notes)) is required for any pull request which modifies:

 - The structure or layout of the release artifacts.
 - The evaluation performance (memory, execution time) of an existing query.
 - The results of an existing query in any circumstance.

If you are only adding new rule queries, a change note is not required.

_**Author:**_ Is a change note required? 
  - [x] Yes
  - [ ] No

🚨🚨🚨
_**Reviewer:**_ Confirm that format of *shared* queries (not the .qll file, the
.ql file that imports it) is valid by running them within VS Code. 
  - [x] Confirmed

_**Reviewer:**_ Confirm that either a change note is not required or the change note is required and has been added.
  - [x] Confirmed

## Query development review checklist

For PRs that add new queries or modify existing queries, the following checklist should be completed by both the author and reviewer:

### Author

 - [x] Have all the relevant rule package description files been checked in?
 - [x] Have you verified that the metadata properties of each new query is set appropriately?
 - [x] Do all the unit tests contain both "COMPLIANT" and "NON_COMPLIANT" cases?
 - [x] Are the alert messages properly formatted and consistent with the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#query-style-guide)?
 - [x] Have you run the queries on OpenPilot and verified that the performance and results are acceptable?<br />_As a rule of thumb, predicates specific to the query should take no more than 1 minute, and for simple queries be under 10 seconds. If this is not the case, this should be highlighted and agreed in the code review process._
 - [x] Does the query have an appropriate level of in-query comments/documentation?
 - [x] Have you considered/identified possible edge cases?
 - [x] Does the query not reinvent features in the standard library?
 - [x] Can the query be simplified further (not golfed!)

### Reviewer 

 - [x] Have all the relevant rule package description files been checked in?
 - [x] Have you verified that the metadata properties of each new query is set appropriately?
 - [x] Do all the unit tests contain both "COMPLIANT" and "NON_COMPLIANT" cases?
 - [x] Are the alert messages properly formatted and consistent with the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#query-style-guide)?
 - [x] Have you run the queries on OpenPilot and verified that the performance and results are acceptable?<br />_As a rule of thumb, predicates specific to the query should take no more than 1 minute, and for simple queries be under 10 seconds. If this is not the case, this should be highlighted and agreed in the code review process._
 - [x] Does the query have an appropriate level of in-query comments/documentation?
 - [x] Have you considered/identified possible edge cases?
 - [x] Does the query not reinvent features in the standard library?
 - [x] Can the query be simplified further (not golfed!)

### CODE REVIEW

File: cpp/autosar/src/rules/A4-7-1/IntegerExpressionLeadToDataLoss.ql [hunk 1]
```diff
@@ -30,5 +30,7 @@ where
   not e instanceof MulExpr and
   // Not covered by this query - overflow/underflow in division is rare
```
Comment (knewbury01) [inline cpp/autosar/src/rules/A4-7-1/IntegerExpressionLeadToDataLoss.ql:31]: quick question, do we need to consider operand types for this query? in the fp report the wording is "regardless of the arguments involved" and its mentioned that the expectation is for integral types to not be an issue, but floating point types still could be?

**Review by knewbury01 — COMMENTED** [review]
one question about the exclusion ask, @rvermeulen

File: cpp/autosar/src/rules/A4-7-1/IntegerExpressionLeadToDataLoss.ql (hunk 1 above)
Comment (rvermeulen) [reply cpp/autosar/src/rules/A4-7-1/IntegerExpressionLeadToDataLoss.ql:31]: This query only targets integers. I will have look at where this is constrainted.

Comment (rvermeulen) [reply cpp/autosar/src/rules/A4-7-1/IntegerExpressionLeadToDataLoss.ql:31]: Already address in PR #490

**Review by knewbury01 — APPROVED** [review]
lgtm!

**Pull request merged**